http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/auth/sam/SamAuthProvider.scala

```diff
@@ -206,107 +206,96 @@ class SamAuthProvider[F[_]: Effect: Logger](samDao: SamDAO[F],
-        .getResourcePolicies[SamNotebookClusterPolicy](authHeader, ResourceTypeName.NotebookCluster)
-      createdPolicies = clusterPolicies.filter(_.accessPolicyName == AccessPolicyName.Creator)
-    } yield clusters.filter {
-      case (project, internalId) =>
-        owningProjects.contains(project) || createdPolicies.exists(_.internalId == internalId)
+      runtimePolicies <- samDao
+        .getResourcePolicies[SamRuntimePolicy](authHeader, SamResourceType.Runtime)
+      createdPolicies = runtimePolicies.filter(_.accessPolicyName == AccessPolicyName.Creator)
+    } yield runtimes.filter {
+      case (project, samResource) =>
+        owningProjects.contains(project) || createdPolicies.exists(_.resource == samResource)
     }
   }
 
-  override def filterUserVisiblePersistentDisks(userInfo: UserInfo, disks: List[(GoogleProject, DiskSamResourceId)])(
+  override def filterUserVisiblePersistentDisks(userInfo: UserInfo,
+                                                disks: List[(GoogleProject, PersistentDiskSamResource)])(
     implicit ev: ApplicativeAsk[F, TraceId]
-  ): F[List[(GoogleProject, DiskSamResourceId)]] = {
+  ): F[List[(GoogleProject, PersistentDiskSamResource)]] = {
     val authHeader = Authorization(Credentials.Token(AuthScheme.Bearer, userInfo.accessToken.token))
     for {
-      projectPolicies <- samDao.getResourcePolicies[SamProjectPolicy](authHeader, ResourceTypeName.BillingProject)
+      projectPolicies <- samDao.getResourcePolicies[SamProjectPolicy](authHeader, SamResourceType.Project)
       owningProjects = projectPolicies.collect {
         case x if (x.accessPolicyName == AccessPolicyName.Owner) => x.googleProject
       }
       diskPolicies <- samDao
-        .getResourcePolicies[SamPersistentDiskPolicy](authHeader, ResourceTypeName.PersistentDisk)
+        .getResourcePolicies[SamPersistentDiskPolicy](authHeader, SamResourceType.PersistentDisk)
       createdPolicies = diskPolicies.filter(_.accessPolicyName == AccessPolicyName.Creator)
     } yield disks.filter {
-      case (project, internalId) =>
-        owningProjects.contains(project) || createdPolicies.exists(_.internalId == internalId)
+      case (project, samResource) =>
+        owningProjects.contains(project) || createdPolicies.exists(_.resource == samResource)
     }
   }
 
-  override def notifyClusterCreated(internalId: RuntimeInternalId,
+  override def notifyRuntimeCreated(samResource: RuntimeSamResource,
                                     creatorEmail: WorkbenchEmail,
-                                    googleProject: GoogleProject,
-                                    clusterName: RuntimeName)(implicit ev: ApplicativeAsk[F, TraceId]): F[Unit] =
-    samDao.createClusterResource(internalId, creatorEmail, googleProject, clusterName)
+                                    googleProject: GoogleProject)(implicit ev: ApplicativeAsk[F, TraceId]): F[Unit] =
+    samDao.createResource(samResource, creatorEmail, googleProject)
```

Comment:
this has the same implementation as `notifyPersistentDiskCreated`, no? also `notifyRuntimeDeleted` and `notifyPersistentDiskDeleted`